core/src/main/scala/com/evolutiongaming/kafka/flow/ConsumerFlow.scala

```diff
@@ -81,21 +81,26 @@ object ConsumerFlow {
         case None         => new IllegalArgumentException("Parameter flows cannot be empty").raiseError[F, Unit]
       }
 
-    def poll = {
+    def poll(logger: Log[F]) = {
       val flowList = flows.toList // optimization, execute toList once instead of on each `consumer.poll`
-      consumer.poll(config.pollTimeout) flatTap { consumerRecords =>
-        flowList traverse { case (topic, flow) =>
-          val topicRecords = consumerRecords.values filter { case (partition, _) =>
-            partition.topic == topic
-          }
-          flow(ConsumerRecords(topicRecords))
+      logger.debug(s"consumer.poll(${config.pollTimeout})") *>
```

Comment:
Let's make it a for-comprehension, please. It's quite hard to read, especially, during the review (see the screenshot). 
<img width="679" alt="Screenshot 2023-01-16 at 1 30 49 PM" src="https://user-images.githubusercontent.com/7479593/212680164-997820e7-ee6d-4e39-ad35-11dfc26054bd.png">
I suggest transforming it to something like this:
```scala
for {
  _ <- logger.debug(s"consumer.poll(${config.pollTimeout})")
  consumerRecords <- consumer.poll(config.pollTimeout)
  _ <- logger.debug(s"polled records: ${ConsumerRecords.summaryShow.show(consumerRecords)}")
  _ <- flowList.traverse { case (topic, flow) =>
    val topicRecords = consumerRecords.values filter { case (partition, _) =>
      partition.topic == topic
    }
    flow(ConsumerRecords(topicRecords))
  }
  _ <- logger.debug(s"poll completed")
} yield consumerRecords
```
This one looks more like straightforward imperative code with fewer levels of nesting.